Number_types/test/Number_types/to_interval_test_boost.cpp

```diff
@@ -796,7 +796,7 @@ void test_to_interval_tight_integer() {
 
 void test_shift_positive() {
   {
-    double d = (1L << 53) - 1;
+    double d = (1LL << 53) - 1;
```

Comment:
You also need to change the others below... (there are warnings if you compile with -m32)